rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java

```diff
@@ -531,10 +553,16 @@ public ResponseEntity<HalResource> addAttachmentToComponent(
             @Parameter(description = "The file to be uploaded.")
             @RequestPart("file") MultipartFile file,
             @Parameter(description = "The attachment info to be created.")
-            @RequestPart("attachment") Attachment newAttachment
+            @RequestPart("attachment") Attachment newAttachment,
+            @Parameter(description = "Comment message.")
+            @RequestParam(value = "comment", required = false) String comment
     ) throws TException {
         final User sw360User = restControllerHelper.getSw360UserFromAuthentication();
         final Component component = componentService.getComponentForUserById(componentId, sw360User);
+        sw360User.setCommentMadeDuringModerationRequest(comment);
+        if (!restControllerHelper.isWriteActionAllowed(component, sw360User) && comment == null) {
+            return new ResponseEntity(RESPONSE_BODY_FOR_MODERATION_REQUEST_WITH_COMMIT, HttpStatus.ACCEPTED);
```

Comment:
In case of missing request parameter, shouldn't the response status be `BAD_REQUEST`?